### ISSUE TITLE
Relax the internal validation checks to allow inlined mock creation in stubbing

### ISSUE THREAD

**Issue by GoogleCodeExporter** [dlsgusrn7577/mockito#191]
```
Relax the internal validation checks to allow inlined mock creation in stubbing
```

Original issue reported on code.google.com by `szcze...@gmail.com` on 24 May 2010 at 8:00

**Comment by GoogleCodeExporter** [comment]
```
Allow both options #1 and #2. Previously, only option #2 was permitted due to 
validation checks.

//1
when(mock.foo()).thenReturn(mock(Foo.class));

//2
Foo foo = mock(Foo.class);
when(mock.foo()).thenReturn(foo);


```

Original comment by `szcze...@gmail.com` on 31 Oct 2010 at 8:30

**Comment by GoogleCodeExporter** [comment]
```
BTW. it's a well known code smell when mocks return mocks so don't get too 
excited about this feature :)
```

Original comment by `szcze...@gmail.com` on 31 Oct 2010 at 8:53

**Comment by GoogleCodeExporter** [comment]
```
This issue was closed by revision 590e8eb051.
```

Original comment by `szcze...@gmail.com` on 31 Oct 2010 at 8:55
- Changed state: **Fixed**

**Comment by GoogleCodeExporter** [comment]
Original comment by `szcze...@gmail.com` on 3 Jul 2011 at 12:43
- Added labels: **Milestone-Release1.9-rc1**
